src/plugins/cmakeprojectmanager/cmaketool.cpp

```diff
@@ -48,6 +48,7 @@ const char CMAKE_INFORMATION_DISPLAYNAME[] = "DisplayName";
 const char CMAKE_INFORMATION_AUTORUN[] = "AutoRun";
 const char CMAKE_INFORMATION_AUTO_CREATE_BUILD_DIRECTORY[] = "AutoCreateBuildDirectory";
 const char CMAKE_INFORMATION_AUTODETECTED[] = "AutoDetected";
+const char CMAKE_INFORMATION_AUTODETECTED_BY_PLUGIN[] = "AutoDetectedByPlugin";
```

Comment:
Why is it necessary to distinguish CMake tools which are autodetected by other plugins? Couldn't we just mark them as AutoDetected?

Reply:
Do you really think I would do it just for fun? :) :) Commit message explains:

> Other plugins may autodetect CMake tools on custom paths. Those would be
> dropped by CMakeToolSettingsAccessor on startup.

Reply:
No I don't :) 
I misread the CMakeToolSettingsAccessor change and was a bit confused by it. Now reading it again I understand the purpose.

Reply:
Yeah it's a bit ugly hack there :) Pity they did not implement it similarly how it is done under Debugger plugin which allows for plugin provided autodetection :\